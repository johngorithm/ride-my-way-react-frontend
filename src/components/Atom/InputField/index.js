import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const InputField = (props) => (
  <Fragment >
    <input {...props} />
  </Fragment>
);

InputField.propTypes = {
  label: PropTypes.string,
  placeholder: PropTypes.string,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired
};

export default InputField;
