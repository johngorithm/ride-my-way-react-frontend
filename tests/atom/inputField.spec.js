import React from 'react';
import { shallow } from 'enzyme';

import InputField from 'components/Atom/InputField';

const props = {
  name: 'password',
  type: 'text',
  value: 'fd89ya8sdf',
  onChange: jest.fn(),
  label: 'Password'
}

describe('Button Tests', () => {
  test('it renders successfully', () => {
    const wrapper = shallow(<InputField {...props} />);
    expect(wrapper.exists()).toBe(true);
  })

  test('it matches snapshot', () => {
    const wrapper = shallow(<InputField {...props}/>);

    expect(wrapper).toMatchSnapshot();
  })
})