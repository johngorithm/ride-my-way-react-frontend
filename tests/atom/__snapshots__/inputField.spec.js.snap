// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests it matches snapshot 1`] = `
<Fragment>
  <input
    label="Password"
    name="password"
    onChange={[MockFunction]}
    type="text"
    value="fd89ya8sdf"
  />
</Fragment>
`;
